Eigen/src/Core/products/GeneralBlockPanelKernel.h

```diff
@@ -90,6 +90,21 @@ struct CacheSizes {
   std::ptrdiff_t m_l3;
 };
 
+// In C++17 this could be an inline variable, see for example
+// https://stackoverflow.com/questions/38043442/how-do-inline-variables-work
+template <typename>
+struct CacheSizeGlobalHelper {
+  static CacheSizes s_cacheSizes;
+};
+#ifdef _OPENMP
+#pragma omp declare target
+#endif
+template <typename T>
+CacheSizes CacheSizeGlobalHelper<T>::s_cacheSizes;
```

Comment:
@olupton : for my understanding - is there a "non-trivial" reason to wrap it as a static member under struct? (apart from global scope variable)

Reply:
I believe this is needed for pre C++17 support (see SO link)

Reply:
Yes, I put [this link](https://stackoverflow.com/questions/38043442/how-do-inline-variables-work) in the comment. With older C++ versions you had to ensure there was only one definition of a global variable in the whole program, usually by putting it in a single `.cpp` file. That's tricky for header-only libraries. (the first draft of this PR gave link errors in the CI because multiple mechanism object files ended up containing definitions -- this is not just a technical/purist's change)

With C++17 you can declare `inline` variables, which are allowed to have multiple identical definitions of which only one will be kept. In C++14 you can achieve something similar with this static member-of-template-struct pattern. It seems to work...

Reply:
I think I know where I was wrong - I was thinking about difference in this case of `pure global variable` vs `static variable wrapped in a struct`. BUT, obviously Eigen is header only library and this variable declared in a header needs to be wrapped-in like this to avoid multiple definitions.

Reply:
Oops..Ok, I read your comment now. Understood.